addons/common/fnc_waitUntilAndExecute.sqf

```diff
@@ -25,8 +28,12 @@ Author:
 ---------------------------------------------------------------------------- */
 #include "script_component.hpp"
 
-params [["_condition", {}, [{}]], ["_statement", {}, [{}]], ["_args", []]];
+params [["_condition", {}, [{}]], ["_statement", {}, [{}]], ["_args", []], ["_timeout", 1E9, [0]], ["_timeoutCode", {}, [{}]]];
 
-GVAR(waitUntilAndExecArray) pushBack [_condition, _statement, _args];
+if (_timeout == 1E9) then {
```

Comment:
Could make default `0` and go for `_timeout > 0`, since 0 whould mean there is an immediate timeout.